core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/orchestration/ClusterProxyRegistrationHandler.java

```diff
@@ -39,18 +52,31 @@ public String selector() {
     @Override
     public void accept(Event<ClusterProxyRegistrationRequest> event) {
         ClusterProxyRegistrationRequest request = event.getData();
-        Selectable response;
+        Selectable response = registerCluster(request);
+        eventBus.notify(response.selector(), new Event<>(event.getHeaders(), response));
+    }
+
+    private Selectable registerCluster(ClusterProxyRegistrationRequest request) {
+        Stack stack = stackService.getByIdWithListsInTransaction(request.getResourceId());
         try {
-            if (clusterProxyIntegrationEnabled) {
-                clusterProxyService.registerCluster(request.getResourceId());
-                response = new ClusterProxyRegistrationSuccess(request.getResourceId());
-            } else {
-                response = new ClusterProxyRegistrationSuccess(request.getResourceId());
+            if (!clusterProxyIntegrationEnabled) {
+                return new ClusterProxyRegistrationSuccess(request.getResourceId());
             }
+            ConfigRegistrationResponse registerResponse = clusterProxyService.registerCluster(stack);
+            Cluster cluster = stack.getCluster();
+            if (cluster.hasGateway()) {
+                LOGGER.debug("Updating Gateway for cluster {} in environment {} with public key certificate retrieved from Cluster Proxy",
+                        cluster.getId(), stack.getEnvironmentCrn());
+                Gateway gateway = cluster.getGateway();
+                gateway.setTokenCert(registerResponse.getX509Unwrapped());
+                gateway.setSsoType(SSOType.SSO_PROVIDER);
```

Comment:
Do we need to update the sso type? I think this comes from the API requests.

Reply:
When debugging locally we didn't see it getting populated - gets persisted as 'NONE'. Was this changed recently?

Reply:
NONE is the default value if you don't provide it on the API. 
```
    "gateway": {
      "topologies": [
        {
          "topologyName": "dp-proxy",
          "exposedServices": [
            "ALL"
          ]
        }
      ],
      "ssoType": "SSO_PROVIDER"
    }
```
SDX and DistroX already sets them to `SSO_PROVIDER`

Reply:
Ok. Looks like this is not getting set from the UI.